src/main/jbake/content/blog/2017/06/9/opentracing-spring-boot.adoc

```diff
@@ -0,0 +1,149 @@
+</dependency>
+----
+
+This dependency requires only one thing and that is tracer bean which will be used to report data to the chosen
+tracing system. If we don't specify this bean auto-configuration will choose `NoopTracer`.
+
+Because we are using OpenTracing instrumentation we are not bound to any specific tracing system. If we now choose for
+example Zipkin it is just a matter of configuration change to switch to a different system for instance Jeager or
+LightStep.
+
+As we mentioned the only tracing configuration needed here is to provide a tracer bean. First we are going to Jaeger and
+then Zipkin tracer.
+
+=== Jaeger
+To create Jaeger tracer is very simple. It just requires sampler configuration and because it is a demo we are going
+to sample all requests. Note that we are not specifying URL to Jaeger server. By default it will assume that it
+runs on the localhost.
+
+----
+@Bean
+public io.opentracing.Tracer jaegerTracer() {
+    return new Configuration("spring-boot", new Configuration.SamplerConfiguration(ProbabilisticSampler.TYPE, 1),
+        new Configuration.ReporterConfiguration())
+        .getTracer();
+----
+
+Now we can start Jaeger server by `docker run --rm -it --network=host jaegertracing/all-in-one`, compile and
+run our app. When everything is up and running generate some requests to URL's defined in the previous section.
+
+Open the Jaeger UI on `http://localhost:16686`:
+
+ifndef::env-github[]
+image::/img/blog/2017/boot-jaeger-traces.png[caption="Figure 1: ", title="Jaeger showing reported traces."]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/boot-jaeger-traces.png[caption="Figure 1: ", title="Jaeger showing
+reported traces."]
+endif::[]
+
+=== Zipkin
+Now let's benefit from OpenTracing and switch tracing system with O(1) effort. To to that we just need to
+provide an instance of Zipkin tracer bean. Do not forget comment out Jaeger tracer bean, otherwise instrumentation
+would not know which tracer to use.
+
+Zipkin configuration is very similar it just requires to know Zipkin URL:
+
+----
+@Bean
+public io.opentracing.Tracer zipkinTracer() {
+    OkHttpSender okHttpSender = OkHttpSender.create("http://localhost:9411/api/v1/spans");
+    AsyncReporter<Span> reporter = AsyncReporter.builder(okHttpSender).build();
+    Tracing braveTracer = Tracing.newBuilder().localServiceName("spring-boot").reporter(reporter).build();
+    return BraveTracer.create(braveTracer);
+}
+----
+
+Zipkin server can be started with `docker run --rm -it -p 9411:9411 openzipkin/zipkin`. Now we have to rebuild and
+start out demo app and generate requests.
+
+ifndef::env-github[]
+image::/img/blog/2017/boot-zipkin-traces.png[caption="Figure 1: ", title="Zipkin showing reported traces."]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/boot-zipkin-traces.png[caption="Figure 1: ", title="Zipkin showing reported traces."]
+endif::[]
+
+== Video
+video::RvCcWltMY7U[youtube]
```

Comment:
preview should be there, I have changed settings to make it a little bit bigger.